xbmc/video/VideoDatabase.cpp

```diff
@@ -8236,13 +8240,21 @@ std::vector<int> CVideoDatabase::CleanMediaType(const std::string &mediaType, co
       SScanSettings scanSettings;
       std::string sourcePath;
       GetSourcePath(parentPath, sourcePath, scanSettings);
+
+      bool bIsSourceName;
+      bool sourceNotFound = (CUtil::GetMatchingSource(parentPath, *videoSources, bIsSourceName) < 0);
+
+      if (sourceNotFound && sourcePath.empty())
+        sourcePath = parentPath;
+
       int sourcePathID = GetPathId(sourcePath);
       std::map<int, std::pair<bool, bool> >::const_iterator sourcePathsDeleteDecision = sourcePathsDeleteDecisions.find(sourcePathID);
       if (sourcePathsDeleteDecision == sourcePathsDeleteDecisions.end())
       {
-        bool sourcePathNotExists = !CDirectory::Exists(sourcePath, false);
+        bool sourcePathNotExists = (!CDirectory::Exists(sourcePath, false) || sourceNotFound);
```

Comment:
The two conditions could be switched because `sourceNotFound` is much cheaper to evaluate.